spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurer.java

```diff
@@ -221,14 +221,18 @@
 
 	private final BeanFactory beanFactory;
 
+	private final RetryTopicNamesProviderFactory retryTopicNamesProviderFactory;
+
 	public RetryTopicConfigurer(DestinationTopicProcessor destinationTopicProcessor,
-						ListenerContainerFactoryResolver containerFactoryResolver,
-						ListenerContainerFactoryConfigurer listenerContainerFactoryConfigurer,
-						BeanFactory beanFactory) {
+								ListenerContainerFactoryResolver containerFactoryResolver,
+								ListenerContainerFactoryConfigurer listenerContainerFactoryConfigurer,
+								BeanFactory beanFactory,
+								RetryTopicNamesProviderFactory retryTopicNamesProviderFactory) {
 		this.destinationTopicProcessor = destinationTopicProcessor;
 		this.containerFactoryResolver = containerFactoryResolver;
 		this.listenerContainerFactoryConfigurer = listenerContainerFactoryConfigurer;
 		this.beanFactory = beanFactory;
+		this.retryTopicNamesProviderFactory = retryTopicNamesProviderFactory;
 	}
```

Comment:
Since this is a public API and we have released 2.7.0 already, we need to keep the old constructor (and deprecate it), populating the new factory with the default.

It is not necessary to do this on the non-public APIs.